state/storage.go

```diff
@@ -120,17 +120,37 @@ func (s *Storage) InsertAccountData(userID, roomID string, events []json.RawMess
 	return data, err
 }
 
+// Prepare a snapshot of the database for calling snapshot functions.
+func (s *Storage) PrepareSnapshot(txn *sqlx.Tx) (tableName string, err error) {
+	// create a temporary table with all the membership nids for the current snapshots for all rooms.
+	// A temporary table will be deleted when the postgres session ends (this process quits).
```

Comment:
It it really when the process ends? Naively I'd've thought it gets dropped when the transaction closes.

Reply:
The reason why we don't use `ON COMMIT DROP` is because that syntax only works with `CREATE TABLE` which we aren't doing here.